[components] Scrapeless - add new actions

joy-chanboop opened the pull request:
## WHY

- Introduced a new README.md for Scrapeless, outlining its features and getting started guide.
- Implemented multiple Scrapeless actions inclues `Crawler`, `Scraping API`, and `Universal Scraping API`.



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced actions for scalable web crawling and scraping, including a Universal Scraping API and a dedicated Scraping API for structured data from popular sites.
  - Added dynamic configuration options for scraping, such as proxy country selection, JavaScript rendering, and headless mode.
  - Provided a new Crawler action for extracting data from single pages or entire domains.

- **Improvements**
  - Enhanced country and region selection with a comprehensive list of over 100 options.
  - Improved error handling and logging for scraping actions.
  - Updated authentication header and API endpoint handling for better reliability.
  - Streamlined input properties and defaults for submitting scrape jobs.

- **Documentation**
  - Added a README introducing Scrapeless, its features, and a getting started guide.

- **Chores**
  - Updated package version and added a new SDK dependency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pipedream-component-development commented:
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

pipedream-component-development commented:
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

YDKD commented on components/scrapeless/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@pipedream/scrapeless",
-  "version": "0.1.0",
+  "version": "0.1.1",
```
Hi, I have already modified the version to `0.2.0` and simplified the request logic. Could you please review it again?

joy-chanboop replied:
I have modified the version to `0.2.0` and simplified the request logic. Could you please review it again?

joy-chanboop commented on components/scrapeless/scrapeless.app.mjs:
```diff
@@ -5,11 +5,11 @@ export default {
   methods: {
     _baseUrl() {
-      return "https://api.scrapeless.com/api/v1";
+      return "https://scrapeless-nodes.norains.com/api/v1";
```
Thank you for your feedback and for pointing out the issue!

Scrapeless is currently integrating with multiple automation platforms (such as Pipedream, n8n, Make, etc.), and we’ve recently released the official [Node SDK](https://www.npmjs.com/package/@scrapeless-ai/sdk). However, during this process, we’ve encountered a key challenge:

> Many platforms do not support adding dependencies as “production dependencies,” which means we cannot directly import and use SDKs like we would in a traditional Node.js environment. Instead, we must rely solely on HTTP requests to interact with services.

In our use case, some services initiate asynchronous tasks, which require multiple steps to complete. This is where the SDK proves helpful, as it abstracts and manages these multi-step processes internally.

To bridge the gap between platforms and provide a unified integration experience, we’ve created a dedicated HTTP service tailored specifically for node-based integrations. That’s why the API base was modified in this PR — to better align with that architecture.

That said, if you believe this change doesn’t align with Pipedream’s best practices, I’m happy to revert to the original approach and implement the polling mechanism you suggested, optimizing it further if needed.

Thanks again for your thoughtful feedback — I really appreciate it and welcome any further suggestions!

joy-chanboop commented:
> Hi @joy-chanboop pending minor change in crawler component!

Hi Jorge Cortes,

I’ve refactored the action calls using the [scrapeless-sdk](https://www.npmjs.com/package/@scrapeless-ai/sdk), and set the official API endpoint as the base URL. I’ve also updated the crawler action key.

Let me know if you have any suggestions or feedback!

joy-chanboop commented:
> Hi @joy-chanboop I've just left a few minor changes! What I usually do to resolve the conflicts in `pnpm-lock.yaml` is to regenerate it with `pnpm install` from the root of the project so here are some steps you can take into consideration:
> 
> ```shell
> $ git pull --rebase
> $ pnpm install
> $ git add pnpm-lock.yaml
> $ git rebase --continue
> $ git push upstream feature/scrapeless-new-actions -f
> ```
Hi @jcortes 
Thank you so much for your guidance earlier. I’ve resolved the conflict, and update your left minor changes.
Would appreciate it if you could kindly review it again when you get a chance. Thanks!

joy-chanboop commented:
> Hi @joy-chanboop,
> 
> Thank you for your information.
> 
> Could you share the API Key to
> 
> * [leo@pipedream.com](mailto:leo@pipedream.com)
> * [sergio@pipedream.com](mailto:sergio@pipedream.com)
> 
> Thanks!

I have sent it to you via email at market@scrapeless.com. Thank you
